Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -476,12 +495,123 @@ void MemoryWidget::OnSearchAddress()
   m_search_offset->setPalette(offset_palette);
 }
 
-void MemoryWidget::ValidateSearchValue()
+void MemoryWidget::ValidateAndPreviewInputValue()
 {
+  m_data_preview->clear();
+  QString input_text = m_data_edit->text();
+
+  if (input_text.isEmpty())
+    return;
+
+  // Remove any spaces
+  if (!m_input_ascii->isChecked())
+    input_text = input_text.replace(QLatin1Char(' '), QStringLiteral(""));
+
   QFont font;
   QPalette palette;
+  QString hex_string;
+  int padding = 8;
+  int hex = 16;
+  bool good = true;
 
-  if (!IsValueValid())
+  if (m_input_ascii->isChecked())
+  {
+    // MemoryView uses Latin1.
+    QByteArray bytes = input_text.toLatin1();
+
+    // An unrestrained size is fine, but it needs to truncate in the preview.
+    if (bytes.size() > 8)
+    {
+      bytes.truncate(8);
+      hex_string = QString::fromLatin1(bytes.toHex()).append(QStringLiteral("...."));
+    }
+    else
+    {
+      hex_string = QString::fromLatin1(bytes.toHex());
+    }
+  }
+  else if (m_input_float->isChecked())
+  {
+    const float float_in = input_text.toFloat(&good);
+
+    if (good)
+    {
+      const u32 hex_out = Common::BitCast<u32>(float_in);
+      hex_string = QStringLiteral("%1").arg(hex_out, padding, hex, QLatin1Char('0')).toUpper();
+    }
+  }
+  else if (m_input_integer->isChecked())
+  {
+    if (input_text.startsWith(QLatin1Char('-')))
+    {
+      int value = input_text.toInt(&good, 10);
+
+      if (good)
+      {
+        // QString turns a negative number into a negative 64-bit number, like -2 ->
+        // 0xFFFFFFFFFFFFFFFE so we remove F's from the left to fit our padding choice.
+        hex_string = QStringLiteral("%1").arg(value, padding, hex, QLatin1Char('0')).toUpper();
+        hex_string.remove(0, 8);
+
+        // If a number is too large, it's upgraded to the next padding size.
+        if (m_type_u8->isChecked() && -128 <= value)
+          hex_string.remove(0, 6);
+        else if ((m_type_u8->isChecked() || m_type_u16->isChecked()) && -32768 <= value)
+          hex_string.remove(0, 4);
+      }
+    }
+    else
+    {
+      u32 value = input_text.toUInt(&good, 10);
+
+      if (good)
+      {
+        // If a number is too large, it's upgraded to the next padding size. Will never be more than
+        // 4 bytes (8 padding), skips being 3 bytes.
+        if (m_type_u8->isChecked() && value <= 0xFF)
+          padding = 2;
+        else if ((m_type_u8->isChecked() || m_type_u16->isChecked()) && value <= 0xFFFF)
+          padding = 4;
+
+        hex_string = QStringLiteral("%1").arg(value, padding, hex, QLatin1Char('0')).toUpper();
+      }
+    }
+  }
+  else if (m_input_hex->isChecked())
+  {
```

Comment:
I often used hex to search for array of bytes so I'm strongly against this `ulonglong` limitation.

Reply:
alright, I wasn't sure about this. I'll work on re-adding the ability.